Don't publish .ts files to npm

I'm getting this error when trying to run `tsc` against an Ember project.

```
node_modules/broccoli-funnel/node_modules/fs-tree-diff/lib/entry.ts:3:1 - error TS1202: Import assignment cannot be used when targeting ECMAScript modules. Consider using 'import * as ns from "mod"', 'import {a} from "mod"', 'import d from "mod"', or another module format instead.

3 import fs = require('fs');
  ~~~~~~~~~~~~~~~~~~~~~~~~~~
```

I believe the problem is that version 2.0.1 was published with both `.js`, `.ts` and `.d.ts` files to npm. Since `index.d.ts` imports from `./entry` and there is an `entry.ts` next to `entry.d.ts`, the former takes precedence. And since it is a `.ts` file, `tsc` uses different rules for checking. For example, `skipLibCheck` only applies to declaration files, and `entry.d.ts` has extra encodings in it based on its `tsconfig.json` that didn't get picked up this way.

Deleting the `*.ts` files from the `fs-tree-diff` folder in `node_modules` fixed the problem.

--- a/package.json
+++ b/package.json
@@ -5,7 +5,8 @@
   "main": "lib/index.js",
   "types": "lib/index.d.ts",
   "files": [
-    "lib"
+    "lib/**/*.js",
+    "lib/**/*.d.ts"
   ],
   "scripts": {
     "test": "npm run test:js",

